azkaban-common/src/main/java/azkaban/project/FlowLoaderFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+* Copyright 2017 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+
+package azkaban.project;
+
+import static java.util.Objects.requireNonNull;
+
+import azkaban.Constants;
+import azkaban.utils.Props;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Factory class to generate flow loaders.
+ */
+@Singleton
+public class FlowLoaderFactory {
```

Comment:
Different flow loader will be created runtime based on the manifest file inside project zip. It would be difficult to guicify it in AzkabanCommonModule.